internal/jobservice/eventstojobs/eventstojobs.go

```diff
@@ -0,0 +1,95 @@
+package eventstojobs
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/G-Research/armada/internal/jobservice/configuration"
+	"github.com/G-Research/armada/internal/jobservice/repository"
+	"github.com/G-Research/armada/pkg/api"
+	"github.com/G-Research/armada/pkg/client"
+)
+
+type EventsToJobService struct {
+	queue                string
+	jobsetid             string
+	jobid                string
+	jobServiceConfig     *configuration.JobServiceConfiguration
+	jobServiceRepository repository.JobServiceRepository
+}
+
+func NewEventsToJobService(
+	queue string,
+	jobsetid string,
+	jobid string,
+	jobServiceConfig *configuration.JobServiceConfiguration,
+	jobServiceRepository repository.JobServiceRepository) *EventsToJobService {
+	return &EventsToJobService{
+		queue:                queue,
+		jobsetid:             jobsetid,
+		jobid:                jobid,
+		jobServiceConfig:     jobServiceConfig,
+		jobServiceRepository: jobServiceRepository}
+}
+
+func (eventToJobService *EventsToJobService) SubscribeToJobSetId(context context.Context) error {
+
+	return eventToJobService.StreamCommon(&eventToJobService.jobServiceConfig.ApiConnection, context)
+}
+func (eventToJobService *EventsToJobService) StreamCommon(clientConnect *client.ApiConnectionDetails, ctx context.Context) error {
+	var fromMessageId string
+	// This corresponds to the amount of time since the job-set has been updated.
```

Comment:
I'm not sure this is what we want.  Rather than unsubscribing when no events have come in for a job set, we want to unsubscribe when nobody has *requested a status update* on that jobset after x hours